### ISSUE TITLE
Unable to build dependencies using requirements.txt

### ISSUE THREAD

**Issue by shahaamirbader** [proroklab/HetGPPO#2]
Great work! Really like your paper. However, When following your instructions and I try running the requirements.txt file the dependencies at torch_scatter just doesnt build and the code is kind of stuck. 

`❯ pip install -r requirements.txt
Looking in indexes: https://pypi.org/simple, https://pypi.ngc.nvidia.com
Requirement already satisfied: numpy==1.23.5 in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from -r requirements.txt (line 1)) (1.23.5)
Collecting vmas
  Downloading vmas-1.2.11.tar.gz (124 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 124.3/124.3 kB 2.1 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
Collecting wandb
  Downloading wandb-0.15.4-py3-none-any.whl (2.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.1/2.1 MB 8.9 MB/s eta 0:00:00
Requirement already satisfied: torch in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from -r requirements.txt (line 4)) (2.0.0)
Collecting torch_geometric
  Downloading torch_geometric-2.3.1.tar.gz (661 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 661.6/661.6 kB 11.0 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting torch_scatter
  Downloading torch_scatter-2.1.1.tar.gz (107 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 107.6/107.6 kB 28.1 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
Collecting torch_sparse
  Downloading torch_sparse-0.6.17.tar.gz (209 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 209.2/209.2 kB 17.2 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
Collecting torch_cluster
  Downloading torch_cluster-1.6.1.tar.gz (53 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 53.8/53.8 kB 75.2 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
Collecting torch_spline_conv
  Downloading torch_spline_conv-1.2.2.tar.gz (25 kB)
  Preparing metadata (setup.py) ... done
Collecting pyglet<=1.5.27
  Downloading pyglet-1.5.27-py3-none-any.whl (1.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.1/1.1 MB 10.0 MB/s eta 0:00:00
Collecting gym
  Downloading gym-0.26.2.tar.gz (721 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 721.7/721.7 kB 10.7 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: six in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from vmas->-r requirements.txt (line 2)) (1.16.0)
Collecting docker-pycreds>=0.4.0
  Downloading docker_pycreds-0.4.0-py2.py3-none-any.whl (9.0 kB)
Requirement already satisfied: Click!=8.0.0,>=7.0 in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from wandb->-r requirements.txt (line 3)) (8.0.4)
Collecting setproctitle
  Downloading setproctitle-1.3.2-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (30 kB)
Requirement already satisfied: appdirs>=1.4.3 in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from wandb->-r requirements.txt (line 3)) (1.4.4)
Requirement already satisfied: protobuf!=4.21.0,<5,>=3.19.0 in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from wandb->-r requirements.txt (line 3)) (4.22.1)
Requirement already satisfied: PyYAML in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from wandb->-r requirements.txt (line 3)) (6.0)
Requirement already satisfied: requests<3,>=2.0.0 in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from wandb->-r requirements.txt (line 3)) (2.28.1)
Requirement already satisfied: setuptools in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from wandb->-r requirements.txt (line 3)) (65.6.3)
Collecting GitPython!=3.1.29,>=1.0.0
  Downloading GitPython-3.1.31-py3-none-any.whl (184 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 184.3/184.3 kB 17.3 MB/s eta 0:00:00
Collecting pathtools
  Downloading pathtools-0.1.2.tar.gz (11 kB)
  Preparing metadata (setup.py) ... done
Collecting sentry-sdk>=1.0.0
  Downloading sentry_sdk-1.25.1-py2.py3-none-any.whl (206 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 206.7/206.7 kB 16.3 MB/s eta 0:00:00
Requirement already satisfied: psutil>=5.0.0 in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from wandb->-r requirements.txt (line 3)) (5.9.0)
Requirement already satisfied: nvidia-cusparse-cu11==11.7.4.91 in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from torch->-r requirements.txt (line 4)) (11.7.4.91)
Requirement already satisfied: nvidia-nccl-cu11==2.14.3 in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from torch->-r requirements.txt (line 4)) (2.14.3)
Requirement already satisfied: jinja2 in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from torch->-r requirements.txt (line 4)) (3.1.2)
Requirement already satisfied: nvidia-nvtx-cu11==11.7.91 in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from torch->-r requirements.txt (line 4)) (11.7.91)
Requirement already satisfied: nvidia-cusolver-cu11==11.4.0.1 in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from torch->-r requirements.txt (line 4)) (11.4.0.1)
Requirement already satisfied: nvidia-curand-cu11==10.2.10.91 in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from torch->-r requirements.txt (line 4)) (10.2.10.91)
Requirement already satisfied: triton==2.0.0 in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from torch->-r requirements.txt (line 4)) (2.0.0)
Requirement already satisfied: sympy in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from torch->-r requirements.txt (line 4)) (1.11.1)
Requirement already satisfied: nvidia-cuda-cupti-cu11==11.7.101 in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from torch->-r requirements.txt (line 4)) (11.7.101)
Requirement already satisfied: nvidia-cudnn-cu11==8.5.0.96 in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from torch->-r requirements.txt (line 4)) (8.5.0.96)
Requirement already satisfied: networkx in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from torch->-r requirements.txt (line 4)) (2.8.4)
Requirement already satisfied: nvidia-cufft-cu11==10.9.0.58 in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from torch->-r requirements.txt (line 4)) (10.9.0.58)
Requirement already satisfied: typing-extensions in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from torch->-r requirements.txt (line 4)) (4.4.0)
Requirement already satisfied: nvidia-cublas-cu11==11.10.3.66 in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from torch->-r requirements.txt (line 4)) (11.10.3.66)
Requirement already satisfied: nvidia-cuda-nvrtc-cu11==11.7.99 in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from torch->-r requirements.txt (line 4)) (11.7.99)
Requirement already satisfied: nvidia-cuda-runtime-cu11==11.7.99 in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from torch->-r requirements.txt (line 4)) (11.7.99)
Requirement already satisfied: filelock in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from torch->-r requirements.txt (line 4)) (3.12.2)
Requirement already satisfied: wheel in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from nvidia-cublas-cu11==11.10.3.66->torch->-r requirements.txt (line 4)) (0.38.4)
Requirement already satisfied: lit in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from triton==2.0.0->torch->-r requirements.txt (line 4)) (16.0.0)
Requirement already satisfied: cmake in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from triton==2.0.0->torch->-r requirements.txt (line 4)) (3.26.1)
Requirement already satisfied: scipy in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from torch_geometric->-r requirements.txt (line 5)) (1.10.0)
Requirement already satisfied: scikit-learn in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from torch_geometric->-r requirements.txt (line 5)) (1.2.1)
Requirement already satisfied: pyparsing in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from torch_geometric->-r requirements.txt (line 5)) (3.0.9)
Requirement already satisfied: tqdm in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from torch_geometric->-r requirements.txt (line 5)) (4.64.1)
Collecting gitdb<5,>=4.0.1
  Downloading gitdb-4.0.10-py3-none-any.whl (62 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 62.7/62.7 kB 85.1 MB/s eta 0:00:00
Requirement already satisfied: charset-normalizer<3,>=2 in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from requests<3,>=2.0.0->wandb->-r requirements.txt (line 3)) (2.0.4)
Requirement already satisfied: certifi>=2017.4.17 in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from requests<3,>=2.0.0->wandb->-r requirements.txt (line 3)) (2022.12.7)
Requirement already satisfied: idna<4,>=2.5 in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from requests<3,>=2.0.0->wandb->-r requirements.txt (line 3)) (3.4)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from requests<3,>=2.0.0->wandb->-r requirements.txt (line 3)) (1.26.14)
Collecting gym-notices>=0.0.4
  Downloading gym_notices-0.0.8-py3-none-any.whl (3.0 kB)
Requirement already satisfied: cloudpickle>=1.2.0 in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from gym->vmas->-r requirements.txt (line 2)) (2.0.0)
Requirement already satisfied: MarkupSafe>=2.0 in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from jinja2->torch->-r requirements.txt (line 4)) (2.1.1)
Requirement already satisfied: joblib>=1.1.1 in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from scikit-learn->torch_geometric->-r requirements.txt (line 5)) (1.1.1)
Requirement already satisfied: threadpoolctl>=2.0.0 in /home/xubuntu/anaconda3/lib/python3.10/site-packages (from scikit-learn->torch_geometric->-r requirements.txt (line 5)) (2.2.0)
Requirement already satisfied: mpmath>=0.19 in /home/xubuntu/anaconda3/lib/python3.10/site-packages/mpmath-1.2.1-py3.10.egg (from sympy->torch->-r requirements.txt (line 4)) (1.2.1)
Collecting smmap<6,>=3.0.1
  Downloading smmap-5.0.0-py3-none-any.whl (24 kB)
Building wheels for collected packages: vmas, torch_geometric, torch_scatter, torch_sparse, torch_cluster, torch_spline_conv, gym, pathtools
  Building wheel for vmas (setup.py) ... done
  Created wheel for vmas: filename=vmas-1.2.11-py3-none-any.whl size=158926 sha256=f1522b764acae4657e82d100e1dfbcc951d2e40bde8cf88fb8827b9368647a78
  Stored in directory: /tmp/pip-ephem-wheel-cache-etc00n1b/wheels/23/ee/97/d1f574c1afa612a0cb0360a70edf39d91bdc652ecde0a43e4e
  Building wheel for torch_geometric (pyproject.toml) ... done
  Created wheel for torch_geometric: filename=torch_geometric-2.3.1-py3-none-any.whl size=910459 sha256=5fd7ba12e5b3af28623df7eb37f19dbf89400c1731cede92ec9bf7f18cfe1d25
  Stored in directory: /tmp/pip-ephem-wheel-cache-etc00n1b/wheels/aa/16/a8/fd7737d723cc1eb8df023c016c262ff4520091e1b022f8c164
  Building wheel for torch_scatter (setup.py) ... |`

**Comment by Gaochengzhi** [comment]
same torch_scatter set up problem, solved by downgrade Nvida driver to 11.8.

**Comment by matteobettini** [comment]
Thanks for posting and sorry I missed this earlier (my notifications were off)

I removed those from the requirements, to intall torch_geometric just follow the guide at https://pytorch-geometric.readthedocs.io/en/latest/install/installation.html
